src/recovery.rs

```diff
@@ -116,77 +607,27 @@ impl fmt::Display for ResumeEpoch {
     }
 }
 
-/// Metadata about an execution.
-///
-/// This represents a row in in the `exs` table.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub(crate) struct ExecutionMeta(ExecutionNumber, WorkerCount, ResumeEpoch);
-
-/// The oldest epoch for which work is still outstanding on a worker.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub(crate) struct WorkerFrontier(u64);
-
-impl fmt::Display for WorkerFrontier {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", self.0)
-    }
-}
-
-/// Metadata about the current frontier of a worker.
-///
-/// This represents a row in the `fronts` table.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub(crate) struct FrontierMeta(ExecutionNumber, WorkerIndex, WorkerFrontier);
-
-/// Metadata about a commit in a recovery partition.
-///
-/// This represents a row in the `commits` table.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub(crate) struct CommitMeta(PartitionIndex, u64);
-
-/// System time duration to keep around state snapshots, even after
-/// they are no longer needed by the current execution.
-///
-/// This is used to delay GC of state data so that if durable backup
-/// of recovery partitions are not instantaneous or synchronized, we
-/// can ensure that there's some resume epoch shared by all partitions
-/// we can use when resuming from a backup that might not be the most
-/// recent.
-#[derive(Debug, Copy, Clone)]
-pub(crate) struct BackupInterval(TimeDelta);
-
-impl Default for BackupInterval {
-    fn default() -> Self {
-        Self(TimeDelta::zero())
-    }
-}
-
-impl IntoPy<Py<PyAny>> for BackupInterval {
-    fn into_py(self, py: Python<'_>) -> Py<PyAny> {
-        self.0.into_py(py)
-    }
-}
-
-impl<'py> FromPyObject<'py> for BackupInterval {
-    fn extract_bound(obj: &Bound<'py, PyAny>) -> PyResult<Self> {
-        if let Ok(duration) = obj.extract::<TimeDelta>() {
-            Ok(Self(duration))
-        } else {
-            Err(PyTypeError::new_err(
-                "backup interval must be a `datetime.timedelta`",
-            ))
-        }
-    }
-}
-
 /// To resume a dataflow execution, you need to know which epoch to
 /// resume for state, but also which execution to label progress data
 /// with.
 ///
 /// This does not define [`Default`] and should only be calculated via
 /// [`ResumeCalc::resume_from`].
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
-pub(crate) struct ResumeFrom(pub(crate) ExecutionNumber, pub(crate) ResumeEpoch);
+pub(crate) struct ResumeFrom(ExecutionNumber, ResumeEpoch);
+
+impl ResumeFrom {
+    pub fn new(ex_num: ExecutionNumber, resume_epoch: ResumeEpoch) -> Self {
+        Self(ex_num, resume_epoch)
+    }
+
+    pub fn execution(&self) -> ExecutionNumber {
+        self.0
+    }
+    pub fn epoch(&self) -> ResumeEpoch {
+        self.1
+    }
+}
```

Comment:
Because while using the fields I always had to go back to the definition to remember which one was the `ExecutionNumber` and which one was the `ResumeEpoch`, so I opted to use named functions to avoid confusion.